src/sentry/event_manager.py

```diff
@@ -745,7 +739,10 @@ def save(self, project_id, raw=False, assume_normalized=False):
         # save the event
         try:
             with transaction.atomic(using=router.db_for_write(Event)):
+                event.set_data(EventDict(data, skip_renormalization=True))
```

Comment:
Why is this needed ? The entire method worked with data which had been assigned from self._data, which is the same assigned to event.data at line 406.

Reply:
Fair point, this was needed since data diverged from event.data.data when we dropped line 406. Reverted these bits back.